### ISSUE TITLE
Do not enough table when parsing sql with UNION ALL

### ISSUE THREAD

**Issue by sonnb270898** [reata/sqllineage#466]
Hi @reata , i am trying to parse the Oracle query below, but the source table i got only b and b1. The expected source table i want is b,c, b1,c1. Can you fix that?

```
(select a
from b
join c on b.id = c.id)
union all
(select a1
from b1
join c1 on b1.id = c1.id)
```

![image](https://github.com/reata/sqllineage/assets/44206043/1cb4984c-a2e1-4b5d-8dd5-7acdc19aac59)

**Python version (available via `python --version`)**
 - 3.9.18
**SQLLineage version (available via `sqllineage --version`):**
 - 1.4.7

**Comment by sonnb270898** [comment]
Oh, when i remove config dialect='oracle', it got the expected result, is it a bug or ? @@

**Comment by reata** [comment]
Closed via #488
